broker-core/src/main/java/io/zeebe/broker/logstreams/processor/TypedStreamProcessor.java

```diff
@@ -159,16 +160,41 @@ public void wrap(
 
     @Override
     public void processEvent() {
-      writer.reset();
-      responseWriter.reset();
+      try {
+
+        writer.reset();
+        responseWriter.reset();
 
-      this.writer.configureSourceContext(streamProcessorId, position);
+        this.writer.configureSourceContext(streamProcessorId, position);
 
-      // default side effect is responses; can be changed by processor
-      sideEffectProducer = responseWriter;
+        // default side effect is responses; can be changed by processor
+        sideEffectProducer = responseWriter;
+
+        eventProcessor.processRecord(
+            position, event, responseWriter, writer, this::setSideEffectProducer);
+      } catch (Exception exception) {
+        if (event.metadata.getRecordType() == RecordType.COMMAND) {
+          sendCommandRejectionOnException(exception);
+        } else if (event.getMetadata().getRecordType() == RecordType.EVENT) {
+          // TODO(#2028) clean up state
+        }
+
+        // re-throw such that stream process controller skips this record
+        throw exception;
+      }
+    }
+
+    private void sendCommandRejectionOnException(Exception exception) {
+      responseWriter.reset();
 
-      eventProcessor.processRecord(
-          position, event, responseWriter, writer, this::setSideEffectProducer);
+      final String errorMessage =
+          String.format(
+              PROCESSING_ERROR_MESSAGE,
+              event,
+              exception.getMessage(),
+              Arrays.toString(exception.getStackTrace()));
```

Comment:
I added this so we as a developer can find the cause of the exception, with help of the stack trace.
The user it self is only able to send us the log, but without the stack trace we might have no clue where the error happen.

Reply:
Can also replace it with a log statement and remove the stacktrace from the response

Reply:
I don't think a user with no access to broker logs would open an issue here; we should direct them to the broker operator, who will be able to diagnose the problem, even if the diagnostic is "open an issue with the Zeebe guys with this stack trace".

In general, I would propose we always approach it like this:

1. Client users should get the information they need to solve issues when possible, otherwise just enough to help the gateway operator diagnose the problem (e.g. timestamp or request to find in the logs).
1. Gateway operators should get the information they need to solve issues when possible, otherwise just enough to help the broker operator diagnose the problem (e.g. timestamp or request to find in the logs)

I expect we would most likely deal with users with access to broker logs here anyway.

Reply:
Yeah agree will change this accordingly.